File: codes/lua/objmeth.lua
local Person = {}
Person.__index = Person

function Person:new(name, byear)    -- The constructor
  return setmetatable({name = name, byear = byear}, Person)
end

function Person:__tostring()
  return "(" .. self.name .. ": " .. self.byear .. ")"
end

local p = Person:new("Karim", 1986)
print("As string: " .. tostring(p))

File: codes/perl/objmeth.pl
#! /usr/bin/perl -w

{
	package Person;

	sub new {
		my $class = shift;
		my $self = {
			_name => shift,
			_byear => shift,
		};
		bless $self, $class;
		return $self;
	}

	use overload  '""' => sub {
		my ($self) = @_;
		return "($self->{_name}: $self->{_byear})"
	}

}

$p = new Person("Karim", 1986);
print "As string: : $p\n";
